source/crud/crud.md

```diff
@@ -702,6 +714,194 @@ class FindOptions {
    */
   let: Optional<Document>;
 }
+
+class FindOneOptions {
```

Comment:
This duplicates a significant amount of options from `FindOptions`. There are two alternatives:
- Have `FindOptions` extend `FindOneOptions` and add options not supported by `findOne`
- Introduce `CommonFindOptions`, have `FindOptions` and `FindOneOptions` extend that class and add their own.

If we want to avoid duplication, I'd prefer option one.

Reply:
I really dislike the duplication here. Here's what pymongo's docs say for find_one:

> All arguments to :meth:`find` are also valid arguments for :meth:`find_one`, although any `limit` argument will be ignored.

Can't we just say something similar here instead of duplicating?

Reply:
Typescript has syntax for this that we use in our driver:

For example:
```ts
type FindOneOptions = Omit<FindOptions, 'timeoutMode'>
```

Not suggesting we use that language specific syntax here, but I think it would be fine to psuedo declare FindOneOptions are equal to FindOptions _except_ for X Y Z

Reply:
@alcaeus What is your thinking here? Seems like we have a preference for not duplicating the options, but as for exactly how to write it, there are a couple of choices here.

Reply:
After thinking about a good solution, I figured that solving this for `find`/`findOne` is only half the story, as there's lots of other duplication (e.g. `UpdateOptions` and `ReplaceOptions`). I filed DRIVERS-3038 to solve the duplication issue and also remove the code blocks: they are unnecessarily prescriptive and prevent us from properly linking to various options in the document. As such, I'd defer making changes to this for the time being.

Reply:
Something for next spec-fest perhaps :) fine by me

Reply:
Can't we just go with one of the proposed simple solutions for now? We can fix the find/findOne duplication in this PR and then use DRIVERS-3038 to fix the remaining issues.

Reply:
Changed to the `Omit` syntax suggested by @nbbeeken.